tutorials/JsonWriterDefaultStream/JsonWriterDefaultStream.java

```diff
@@ -0,0 +1,126 @@
+  public static void createDestinationTable(
+      String projectId, String datasetName, String tableName) {
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    // Create a schema that matches the source data.
+    Schema schema =
+        Schema.of(
+            Field.of("commit", StandardSQLTypeName.STRING),
+            Field.newBuilder("parent", StandardSQLTypeName.STRING)
+                .setMode(Field.Mode.REPEATED)
+                .build(),
+            Field.of("author", StandardSQLTypeName.STRING),
+            Field.of("committer", StandardSQLTypeName.STRING),
+            Field.of("time_sec", StandardSQLTypeName.INT64),
+            Field.of("subject", StandardSQLTypeName.STRING),
+            Field.of("message", StandardSQLTypeName.STRING),
+            Field.of("repo_name", StandardSQLTypeName.STRING));
+
+    // Create a table that uses this schema.
+    TableId tableId = TableId.of(projectId, datasetName, tableName);
+    Table table = bigquery.getTable(tableId);
+    if (table == null) {
+      TableInfo tableInfo =
+          TableInfo.newBuilder(tableId, StandardTableDefinition.of(schema)).build();
+      bigquery.create(tableInfo);
+    }
+  }
+
+  // writeToDefaultStream: Writes records from the source file to the destination table.
+  public static void writeToDefaultStream(
+      String projectId, String datasetName, String tableName, String dataFile)
+      throws DescriptorValidationException, InterruptedException, IOException {
+
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+    // Get the schema of the destination table and convert to the equivalent BigQueryStorage type.
+    Table table = bigquery.getTable(datasetName, tableName);
+    Schema schema = table.getDefinition().getSchema();
+    TableSchema tableSchema = BqToBqStorageSchemaConverter.convertTableSchema(schema);
+
+    // Use the JSON stream writer to send records in JSON format.
+    TableName parentTable = TableName.of(projectId, datasetName, tableName);
+    try (JsonStreamWriter writer =
+        JsonStreamWriter.newBuilder(parentTable.toString(), tableSchema).build()) {
+      // Read JSON data from the source file and send it to the Write API.
+      BufferedReader reader = new BufferedReader(new FileReader(dataFile));
+      String line = reader.readLine();
+      while (line != null) {
+        // As a best practice, send batches of records, instead of single records at a time.
+        JSONArray jsonArr = new JSONArray();
+        for (int i = 0; i < 100; i++) {
+          JSONObject record = new JSONObject(line);
+          jsonArr.put(record);
+          line = reader.readLine();
+          if (line == null) {
+            break;
+          }
+        } // batch
+        ApiFuture<AppendRowsResponse> future = writer.append(jsonArr);
+        AppendRowsResponse response = future.get();
+      }
+      System.out.println("Appended records successfully.");
```

Comment:
Maybe finalize the stream to report the row count?

Reply:
cc @VeronicaWasson

Reply:
I'm not sure how to finalize the default stream. (We don't finalize the stream in the existing WriteToDefaultStream snippet.)